test_p_bfgs fails on >60 core CPU

<!-- ⚠️ If you do not respect this template, your issue will be closed -->
<!-- ⚠️ Make sure to browse the opened and closed issues -->

### Information

- **Qiskit Aqua version**: 0.7.5
- **Python version**: [3.7, 3.8]
- **Operating system**: Linux

### What is the current behavior?

When repackaging ``qiskit-aqua`` for NixOS, the following error test fails.
```
=================================== FAILURES ===================================
__________________________ TestOptimizers.test_p_bfgs __________________________

self = <test.aqua.test_optimizers.TestOptimizers testMethod=test_p_bfgs>

    def test_p_bfgs(self):
        """ parallel l_bfgs_b test """
        optimizer = P_BFGS(maxfun=1000)
        res = self._optimize(optimizer)
>       self.assertLessEqual(res[2], 10000)
E       AssertionError: 44250 not less than or equal to 10000

test/aqua/test_optimizers.py:69: AssertionError
```

Ref: https://github.com/NixOS/nixpkgs/pull/95713#pullrequestreview-472682769

### Steps to reproduce the problem

1. Get a Threadripper 3990X (or equivalent many-core CPU), running Linux
2. Checkout qiskit-aqua at v0.7.5
3. Run pytest (or stestr)

(Optional, exact reproduction of build):
1. ``git clone https://github.com/drewrisinger/nur-packages``
2. ``git checkout 30b017dfe580c496452e95caa089e56e694ca125``
3. Install nix (Linux only): ``curl -L https://nixos.org/nix/install | sh``
4. Build ``qiskit-aqua``:  ``nix-build -A python3Packages.qiskit-aqua``


### What is the expected behavior?

Test https://github.com/Qiskit/qiskit-aqua/blob/32595efa34f54f55555ede886e38db9ee87c4344/test/aqua/test_optimizers.py#L65-L69 should pass.

### Suggested solutions

It looks like this is failing because the P_BFGS algorithm distributes the optimization task to more cores than the test expected, and then the number of evaluations is too high even if it finds a solution quickly. Maybe you could either raise the number of evaluations or have the evaluations depend on the number of CPU cores somehow?

--- a/test/aqua/test_optimizers.py
+++ b/test/aqua/test_optimizers.py
@@ -64,7 +64,7 @@ class TestOptimizers(QiskitAquaTestCase):
 
     def test_p_bfgs(self):
         """ parallel l_bfgs_b test """
-        optimizer = P_BFGS(maxfun=1000)
+        optimizer = P_BFGS(maxfun=1000, max_processes=4)
         res = self._optimize(optimizer)
         self.assertLessEqual(res[2], 10000)
 

